compiler/rustc_builtin_macros/Cargo.toml

```diff
@@ -3,9 +3,9 @@ name = "rustc_builtin_macros"
 version = "0.0.0"
 edition = "2024"
 
-
-[lints.rust]
-unexpected_cfgs = { level = "warn", check-cfg = ['cfg(llvm_enzyme)'] }
+# njn: is this needed?
+# [lints.rust]
+# unexpected_cfgs = { level = "warn", check-cfg = ['cfg(llvm_enzyme)'] }
```

Comment:
I don't know if this needs to be here. There is a `cfg!(llvm_enzyme)` test within the crate, but I don't know how this lint would be triggered.

Reply:
I would like to, but if you inherit workspace lints with `[lints] workspace = true` you can't add extra lints in the `Cargo.toml`, see https://github.com/rust-lang/cargo/issues/13157. The suggested workaround is to instead use lint attributes in the code. Which I'm happy to do, but I don't know how to translate this lint (in particular the `check-cfg`) into a lint attribute. Maybe it's just a matter of adding `#[allow(unexpected_cfgs)]` on the single `cfg!(llvm_enzyme)` usage in the code? But I can't reproduce the warning either locally or on CI, so I can't test such a change. Help!

Reply:
The cfg is already in the extra check-cfg list in bootstrap, I'm pretty sure you can just remove it as bootstrap already adds it, which I think is clearly visible by the fact that CI isn't complaining about missing it.

https://github.com/rust-lang/rust/blob/03eb45452305f2d52348279d0caa5fc1f12c438d/src/bootstrap/src/lib.rs#L80-L82

Reply:
Aha! That makes sense. Especially because there are some `cfg!(llvm_enzyme)` conditions in a few other crates that don't cause problems. Thank you.